docs/source/developers/java/building.rst

```diff
@@ -132,7 +132,12 @@ Maven
         $ cd arrow/java
         $ export JAVA_HOME=<absolute path to your java home>
         $ java --version
-        $ mvn generate-resources -Pgenerate-libs-jni-macos-linux -N
+        $ mvn generate-resources \
+            -Pgenerate-libs-jni-macos-linux \
+            -DARROW_GANDIVA=ON \
+            -DARROW_JAVA_JNI_ENABLE_GANDIVA=ON \
+            -DARROW_JAVA_JNI_ENABLE_C=ON \
```

Comment:
Should we add this?
This item is described as "To build all JNI libraries (MacOS / Linux) except the JNI C Data Interface library".

Reply:
How about if we change the description? Without this change the build always fails for me.

Reply:
I'm OK with changing the description. (We need to update "To build only the JNI C Data Interface library (MacOS / Linux)" and related items too with the approach.)

@davisusanibar What do you think about this?

BTW, if you run the "To build only the JNI C Data Interface library (MacOS / Linux)" item separately, can we build without error?

Reply:
> I'm OK with changing the description. (We need to update "To build only the JNI C Data Interface library (MacOS / Linux)" and related items too with the approach.)

Ok, I'll push a change.

> BTW, if you run the "To build only the JNI C Data Interface library (MacOS / Linux)" item separately, can we build without error?

```
mvn clean install
mvn generate-resources -Pgenerate-libs-cdata-all-os -N
mvn -Darrow.c.jni.dist.dir=$PWD/../java-dist/lib -Parrow-c-data clean install
```
builds fine on master on my M1.

Reply:
Ah, sorry. I meant the followings:

```bash
mvn clean install
rm -rf ../java-dist ../java-jni
mvn generate-resources -Pgenerate-libs-cdata-all-os -N
mvn generate-resources -Pgenerate-libs-jni-macos-linux -DARROW_GANDIVA=ON -DARROW_JAVA_JNI_ENABLE_GANDIVA=ON -N
mvn -Darrow.cpp.build.dir=$PWD/../java-dist/lib/ -Darrow.c.jni.dist.dir=$PWD/../java-dist/lib/ -Parrow-jni clean install
```

Reply:
These steps pass for me on this branch and fail on master (on `generate-libs-jni-macos-linux` step).

Reply:
Then, how about removing the `-DARROW_JAVA_JNI_ENABLE_C=ON` from this branch?
Building `libarrow_cdata_jni.dylib` and other `.dylib`s for JNI separately is the story used in the current document.

Reply:
```suggestion
```

Reply:
Sure! Removed.